source/RialtoGStreamerMSEVideoSink.cpp

```diff
@@ -46,15 +48,28 @@ enum
 
 static GstStateChangeReturn rialto_mse_video_sink_change_state(GstElement *element, GstStateChange transition)
 {
-    RialtoMSEBaseSink *sink = RIALTO_MSE_BASE_SINK(element);
-    RialtoMSEBaseSinkPrivate *priv = sink->priv;
+    RialtoMSEVideoSink *sink = RIALTO_MSE_VIDEO_SINK(element);
+    RialtoMSEVideoSinkPrivate *priv = sink->priv;
+    RialtoMSEBaseSinkPrivate *basePriv = sink->parent.priv;
 
     switch (transition)
     {
     case GST_STATE_CHANGE_READY_TO_PAUSED:
     {
+        // Attach the media player client to media player manager.
+        // maxWidth and maxHeight are used to set the video capabilities of the MediaPlayer.
+        // If the mediaPlayer has already been created (ie. an audio sink on the same parent bus changed state first)
+        // the video capabilities will NOT be set.
+        if (!basePriv->m_mediaPlayerManager.attachMediaPlayerClient(rialto_mse_base_get_oldest_gst_bin_parent(element),
+                                                                    priv->maxWidth, priv->maxHeight))
+        {
+            GST_ERROR_OBJECT(sink, "Cannot attach the MediaPlayerClient");
+            return GST_STATE_CHANGE_FAILURE;
+        }
+
```

Comment:
Can you add info log here? Something like:
```
GstObject parentObject = rialto_mse_base_get_oldest_gst_bin_parent(element);
if (!basePriv->m_mediaPlayerManager.attachMediaPlayerClient(parentObject, priv->maxWidth, priv->maxHeight))
    {
        GST_ERROR_OBJECT(sink, "Cannot attach the MediaPlayerClient");
        return GST_STATE_CHANGE_FAILURE;
    }
    else
   {
       GST_INFO_OBJECT(element, "Attached media player client with parent %s(%p)", gst_object_get_name(parentObject), parentObject )
   }
```

Reply:
Done